### ISSUE TITLE
Move Dropout to LowMemDropout Replacement To PyDispatcher

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#93603]
Dynamo tracing will replace [dropout](https://github.com/pytorch/pytorch/blob/master/torch/_inductor/overrides.py#L295) with [LowMemDropout](https://github.com/pytorch/pytorch/blob/master/torch/_inductor/overrides.py#L247). Because we are doing it at the bytecode level, this replacment is fragile, and does not work for `nn.Dropout`. We should instead use a pre-autograd pydispatcher decomposition to do the replacement. We should also do this for the `rand_like` replacement.

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh

**Comment by ezyang** [comment]
I guess this is still valid. @jansel @ngimel I wonder what proportion of inductor's pre-autograd optimizations can be done on-the-fly decompositions

**Comment by jansel** [comment]
Any of the ones that match a single op could be.  There are some that touch multiple ops though.

**Comment by ydwu4** [comment]
Hi @eellison, is this issue still valid? Can we close it if it's not interesting anymore?

**Comment by eellison** [comment]
I had a pr open for it, but it regressed compile time. I think
